### DISCUSSION TITLE
Great App Praise

### DISCUSSION THREAD

**Discussion by pawell2418** [yitsushi/totp-cli#84] [General]
I don't know if you already know or not, but recent change in GitHub security policy will make your project probably one of the most popular ones on the website. Instead of installing spyware on your smartphone you just grab your binary and you are done. **Good job!**

My only gripe with the app is the inconsistent access to accounts in various commands:
```
namespace.account
namespace account
```
The size of binary (but that is probably the downside of using Go framework).
And the fact how aggressively the app is demanding password. Oh well at least it doesn't require minimum length or complexity.

Still, good job!

**Comment by yitsushi** [comment]
> My only gripe with the app is the inconsistent access to accounts in various commands:

For a while I keep that for backwards compatibility as I expect a lot of users are using it as part of their CD/CI, so they can update the binary but don't have to change the pipeline itself.

I think I removed that recently and all commands use the `namespace account` format. Even the readme was updated to remove all those references.

```
   totp-cli add-token [command options] [namespace] [account]
   totp-cli generate [command options] <namespace> <account>
   totp-cli set-prefix [command options] [namespace] [account] [prefix]
   totp-cli list [command options] [namespace]
```

**Comment by yitsushi** [comment]
> The size of binary (but that is probably the downside of using Go framework).

Yes, I can't do much about it.

**Reply by yitsushi** [reply]
With `v1.5.3` I tried to reduce the binary a bit, didn't help much, but more than nothing.

```
❯ /usr/bin/du * | awk 'BEGIN { old=0; new=0 } { if ($2 == "totp-cli-v1.5.0") { old=$1 }; if ($2 == "totp-cli-v1.5.3") { new=$1 }; } END { print new/old }'
0.699132

❯ /usr/bin/du -sh *
11M	totp-cli-v1.5.0
5.8M	totp-cli-v1.5.0-linux-amd64.tar.gz
7.3M	totp-cli-v1.5.3
3.1M	totp-cli-v1.5.3-linux-amd64.tar.gz

❯ /usr/bin/du totp-cli-v1.5.0 totp-cli-v1.5.3
10596	totp-cli-v1.5.0
7408	totp-cli-v1.5.3

❯ bc <<< "scale=4; 7408/10596"
.6991
```

https://github.com/yitsushi/totp-cli/releases/tag/v1.5.3

**Comment by yitsushi** [comment]
>  And the fact how aggressively the app is demanding password. Oh well at least it doesn't require minimum length or complexity.

Yes, because without password the file would contain all tokens plain text on the hard drive. [There are several ways to specify the password](https://github.com/yitsushi/totp-cli#about-the-password), so technically you can set up an environment where it never asks for password.

**Comment by yitsushi** [comment]
> Still, good job!

And, thank you :heart: 

(I broke down my answers in separate comments to make it easier to respond to a specific one)
